docs/how-it-works.md

```diff
@@ -0,0 +1,205 @@
+
+### Request Headers
+
+In [Swagger][] documents, HTTP headers are inferred from [`"in": "header"` parameters][parameters]. HTTP headers significant for content negotiation are inferred according to following rules:
+
+- [`consumes`][] is propagated as request's `Content-Type` header.
+- [`produces`][] is propagated as request's `Accept` header.
+
+### Request Body
+
+#### API Blueprint
+
+The effective request body is taken from following places (the order goes from the highest priority to the lowest):
+
+1. [`+ Body`][] section with sample JSON payload.
+2. [`+ Attributes`][] section with data structure description in [MSON][] - API Blueprint parser automatically generates sample JSON payload from MSON.
+
+This order [exactly follows the API Blueprint specification][body-schema-attributes].
+
+#### Swagger
+
+The effective request body is inferred from [`"in": "body"` and `"in": "form"` parameters][parameters]. Since Swagger doesn't provide any way to specify raw JSON sample for the request body, Dredd's [Swagger Adapter][] generates some sample values from the JSON Schema provided in the [`schema`][] property. Following rules apply when the adapter fills values of the properties, ordered by precedence:
+
+1. Value of `default`.
+2. First value from `enum`.
+3. Dummy, generated value.
+
+## How Dredd Works With HTTP Transactions
+
+### Multiple Requests and Responses
+
+#### API Blueprint
+
+While [API Blueprint][] allows specifying multiple requests and responses in any
+combination (see specification for the [action section][]), Dredd currently
+supports just separated HTTP transaction pairs like this:
+
+```
++ Request
++ Response
+
++ Request
++ Response
+```
+
+In other words, Dredd always selects just the first response for each request.
+
+> **Note:** Improving the support for multiple requests and responses is under development. Refer to issues [#25](https://github.com/apiaryio/dredd/issues/25) and [#78](https://github.com/apiaryio/dredd/issues/78) for details. Support for URI parameters specific to a single request within one action is also limited. Solving [#227](https://github.com/apiaryio/dredd/issues/227) should unblock many related problems. Also see [Multiple Requests and Responses within One API Blueprint Action](how-to-guides.md#multiple-requests-and-responses-within-one-api-blueprint-action) guide for workarounds.
+
+#### Swagger
+
+The [Swagger][] format allows to specify multiple responses for a single operation.
+By default Dredd tests only responses with `2xx` status codes. Responses with other
+codes are marked as _skipped_ and can be activated in [hooks](hooks.md) - see [Testing non-2xx Responses with Swagger](how-to-guides.md#testing-non-2xx-responses-with-swagger).
+
+[Default responses][default responses] are ignored by Dredd. Also, as of now,
+only `application/json` media type is supported in [`produces`][] and [`consumes`][].
```

Comment:
This should be extended for `whatever+json` media types and for support of `application/json; whatever`. In other words, to support JSON suffixed media types and omit the parameters.

Consider using this npm: https://www.npmjs.com/package/media-typer